keps/1145-additional-labels/README.md

```diff
@@ -0,0 +1,153 @@
+# KEP-1145: Additional labels, annotations, tolerations and selector
+
+<!-- toc -->
+- [Summary](#summary)
+- [Motivation](#motivation)
+  - [Goals](#goals)
+  - [Non-Goals](#non-goals)
+- [Proposal](#proposal)
+  - [User Stories (Optional)](#user-stories-optional)
+    - [Story 1](#story-1)
+    - [Story 2](#story-2)
+  - [Risks and Mitigations](#risks-and-mitigations)
+- [Design Details](#design-details)
+  - [Test Plan](#test-plan)
+      - [Prerequisite testing updates](#prerequisite-testing-updates)
+    - [Unit Tests](#unit-tests)
+    - [Integration tests](#integration-tests)
+  - [Graduation Criteria](#graduation-criteria)
+- [Implementation History](#implementation-history)
+- [Drawbacks](#drawbacks)
+- [Alternatives](#alternatives)
+<!-- /toc -->
+
+## Summary
+
+Allow (AdmissionCheck)[https://github.com/kubernetes-sigs/kueue/tree/main/keps/993-two-phase-admission] 
+to set some additional location/admission information
+on workload's pods, so that they land on the machines of the `AdmissionCheck`
+choosing.
+
+## Motivation
+
+`AdmissionChecks` is a "plugin" mechanism for Kueue. Some of the plugins while 
+admitting may either narrow down on which machines the workload should start (by adding
+some extra node selector/tolerations) or add some tracking information
+that may be needed by other system components outside of Kueue.
+
+### Goals
+
+* Establish a mechanism through which `AdmissionChecks` can add annotations/labels/node selector/tolerations
+  to workload pods.
+
+### Non-Goals
+
+* Allow `AdmissionChecks` to change pod definition in other way, for instance change command or image name.
+* Hard code any specific labels/annotations that are passed down.
+
+## Proposal
+
+* Add a struct in `Workload` status that allows to set annotations, labels, selectors and tolerations on PodSets.
+* Expend PodSetInfo in Integration Framework to pass this information.
+* Modify existing integrations to support setting more data on the pods.
+
+### User Stories (Optional)
+
+#### Story 1
+
+My external controller managing an `AdmissionCheck` preallocates specific
+nodes (tainted and labeled) for a Workload to run, before it is 
+admitted (and creates any worker pods). Workload pods need to be modified
+so that they end up on correct nodes.
+
+#### Story 2
+
+My `AdmissionCheck` does budget checking and would like to mark pods that are possibly crossing budget with a 
+specific label for easier tracking and alerting.
+
+### Risks and Mitigations
+
+Some integrations may not support setting all of the desired information due to their CRD limitations.
+
+## Design Details
+
+Modify WorkloadStatus API object:
+
+```
+// WorkloadStatus defines the observed state of Workload
+type WorkloadStatus struct {
+    [...]
+    AdmissionChecks []metav1.Condition `json:"admissionChecks,omitempty" patchStrategy:"merge" patchMergeKey:"type"`
+
+    PodSetUpdates []PodSetUpdate `json:"podSetUpdates,omitempty"
```

Comment:
@mimowo @mwielgus this cold be another reason to change  `AdmissionChecks []metav1.Condition ` to a custom type that also contains the `PodSetUpdate` fields. (this being being in the context of the condition naming discussion we had in #1113  )

Reply:
Yes, we could put `PodSetUpdates[]` in the forked condition. It would give a little bit of extra clarity about who wants to set what, but at the cost of redoing all the utility functions for conditions.

Reply:
There are mostly only two utility functions  (SetStatusConditions..., IsStatusCond... ) we can-provide our own.

Reply:
+1 on not using metav1.Condition and creating our own. The way we are already using the `reason` field is odd enough.

Also, in the existing proposal, it's not clear how multiple AdmissionChecks would interact to each provide a modification to the template.

Reply:
Ok, modified.